libs/wingsdk/src/http/http.ts

```diff
@@ -271,6 +327,61 @@ export class Util {
     });
   }
 
+  /**
+   * Parses the input URL String using WHATWG URL API and returns an URL Struct.
+   * @param urlString The URL String to be parsed.
+   * @throws Will throw an error if the input String is not a valid URL.
+   * @inflight
+   * @returns An URL Struct.
+   */
+  public static parseUrl(urlString: string): Url {
+    try {
+      const nodeUrl = new NodeUrl(urlString);
+      return {
```

Comment:
Why not just return nodeUrl as is?

Reply:
https://winglang.slack.com/archives/C048QCN2XLJ/p1695858169336869

Reply:
I was getting this when I tried: `error JSII3001: Type "URL" cannot be used as the return type because it is private or @internal`

@MarkMcCulloh confirmed that you cannot publicly expose types from Node since they aren't from a JSII module.

Reply:
I wasn't suggesting to expose the `URL` type, just to return `nodeUrl as Url`.

Reply:
Oh, we can't do that because `Url` doesn't implement the full `nodeUrl` interface. Specifically, it lacks the `searchParams` object and the methods `toJSON()` and `toString()`.